TLS certificate path should not be based on present working directory

sfc-gh-sgwydir opened the pull request:
I found it surprising that if I had a file named `fdb.pem` in my current directory that this pem file would be used by `fdbcli` rather than the default `/etc/foundationdb/fdb.pem`. This change makes it so that `/etc/foundationdb/fdb.pem` is the default unless another certificate is explicitly specified.

# Code-Reviewer Section

The general guidelines can be found [here](https://github.com/apple/foundationdb/wiki/FoundationDB-Commit-Process).

Please check each of the following things and check *all* boxes before accepting a PR.

- [ ] The PR has a description, explaining both the problem and the solution.
- [ ] The description mentions which forms of testing were done and the testing seems reasonable.
- [ ] Every function/class/actor that was touched is reasonably well documented.

## For Release-Branches

If this PR is made against a release-branch, please also check the following:

- [ ] This change/bugfix is a cherry-pick from the next younger branch (younger `release-branch` or `main` if this is the youngest branch)
- [ ] There is a good reason why this PR needs to go into a release branch and this reason is documented (either in the description above or in a linked GitHub issue)

sfc-gh-etschannen commented:
please pull in main

sfc-gh-jshim reviewed (changes requested):
LGTM, but looks like this warrants a matching [documentation change](https://apple.github.io/foundationdb/tls.html#certificate-file-default-location), as it explicitly says it'll look inside `$PWD`.

Also I'd like to see it pass the CI build.

sfc-gh-jshim commented:
Also, now that we're revisiting `fdb.pem`, it makes more sense to distinguish the key file and the cert file, especially now that we have a `/etc/foundationdb` directory on top of the files. `cert.pem` and `key.pem` seem like good alternatives.

sfc-gh-sgwydir commented:
Rebased onto main and documented the behavior change. Broke `fdb.pem` down into `cert.pem` and `key.pem` and added that they can optionally be combined.

sfc-gh-jshim reviewed (approved):
LGTM! Thanks for doing this.
